File: Hinweis Java/java/src/main/java/de/thm/mni/compilerbau/absyn/VariableDefinition.java
package de.thm.mni.compilerbau.absyn;

import de.thm.mni.compilerbau.absyn.visitor.Visitor;
import de.thm.mni.compilerbau.table.Identifier;

/**
 * This class represents the local declaration of a variable.
 * <p>
 * Variables are defined inside a procedure and combine an {@link Identifier} with a {@link TypeExpression},
 * expressing the variables type.
 * Variables are only visible in the local scope of their procedure.
 */
public class VariableDefinition extends Node {
    public final Identifier name;
    public final TypeExpression typeExpression;

    /**
     * Creates a new node representing the declaration of a local variable in a procedures body.
     *
     * @param position       The position of the declaration in the source code.
     * @param name           The identifier of the defined local variable.
     * @param typeExpression The type expression used to express the type of the local variable.
     */
    public VariableDefinition(Position position, Identifier name, TypeExpression typeExpression) {
        super(position);
        this.name = name;
        this.typeExpression = typeExpression;
    }

    @Override
    public String toString() {
        return formatAst("VariableDefinition", name, typeExpression);
    }

    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }
}

File: Skeleton Kotlin/kotlin/src/main/kotlin/de/thm/mni/compilerbau/absyn/VariableDefinition.kt
package de.thm.mni.compilerbau.absyn

import de.thm.mni.compilerbau.absyn.visitor.Visitor
import de.thm.mni.compilerbau.table.Identifier

/**
 * This class represents the local definition of a variable.
 *
 * Variables are declared inside a procedure and combine an [Identifier] with a [TypeExpression],
 * expressing the variables type.
 * Variables are only visible in the local scope of their procedure.
 *
 * @param position       The position of the definition in the source code.
 * @param name           The identifier of the declared local variable.
 * @param typeExpression The type expression used to express the type of the local variable.
 */
class VariableDefinition(position: Position, val name: Identifier, val typeExpression: TypeExpression) : Node(position) {
    override fun accept(visitor: Visitor) = visitor.visit(this)

    override fun toString(): String = formatAst("VariableDefinition", name, typeExpression)
}
